easybuild/framework/easyconfig/types.py

```diff
@@ -116,3 +110,34 @@ def convert_value_type(val, typ):
         raise EasyBuildError("No conversion function available (yet) for target type %s", typ)
 
     return res
+
+def to_dict(tc):
+    """
+    Convert a toolchain string "foo, v0" to a dictionary {'name':'foo', 'version':'v0'}
+
+    @param toolchain_str: a toolchain string
+    """
+    # check if tc is a string or a list of two values; else, it can not be converted
+    if isinstance(tc, basestring):
+        if tc.count(',') != 1:
+            raise EasyBuildError("Can not convert string %s to toolchain dict. Expected format: name,version" % tc)
+        else:
+            tc_split = tc.split(',')
+            return {'name':tc_split[0].strip(), 'version':tc_split[1].strip()}
+
+    elif isinstance(tc, list):
```

Comment:
avoid duplicate code; use `split` if it's a string originally, and then require that's it's a list for the remainder; that way, you only need to check the length once

```python
if isinstance(tcspec, basestring):
    tcspec = tcspec.split(',')

if isinstance(tcspec, list):
    if len(tcspec) == 2:
      ...
    else:
      raise ...

else:
      raise ...